src/org/zaproxy/zap/extension/replacer/ExtensionReplacer.java

```diff
@@ -0,0 +1,263 @@
+                @Override
+                public void actionPerformed(java.awt.event.ActionEvent ae) {
+                    Control.getSingleton().getMenuToolsControl().options(OptionsReplacerPanel.PANEL_NAME);
+                }
+            });
+        }
+        return replacerMenuItem;
+    }
+
+    private boolean contains (String original, String match, Pattern p) {
+        if (p != null) {
+            return p.matcher(original).find();
+        } else {
+            return original.contains(match);
+        }
+    }
+
+    private String replace (String original, String match, Pattern p, String replacement) {
+        if (p != null) {
+            return p.matcher(original).replaceAll(replacement);
+        } else {
+            return original.replace(match, replacement);
+        }
+    }
+    
+    @Override
+    public void onHttpRequestSend(HttpMessage msg, int initiator, HttpSender httpSender) {
+        for (ReplacerParamRule rule : this.getParams().getRules()) {
+            if (rule.isEnabled() && rule.appliesToInitiator(initiator)) {
+                Pattern p = null;
+                if (rule.isMatchRegex()) {
+                    p = Pattern.compile(rule.getMatchString());
+                }
+                switch (rule.getMatchType()) {
+                case req_header:
+                    LOGGER.debug("Add in request header: " + rule.getMatchString() + " : " + rule.getReplacement());
+                    msg.getRequestHeader().setHeader(rule.getMatchString(), rule.getReplacement());
+                    break;
+                case req_header_str:
+                    LOGGER.debug("Replace in request header: " + rule.getMatchString() + " with " + rule.getReplacement());
+                    String header = msg.getRequestHeader().toString();
+                    if (contains(header, rule.getMatchString(), p)) {
+                        header = replace(header, rule.getMatchString(), p, rule.getReplacement());
+                        try {
+                            msg.setRequestHeader(new HttpRequestHeader(header));
+                        } catch (HttpMalformedHeaderException e) {
+                            LOGGER.error(e.getMessage(), e);
+                        }
+                    }
+                    break;
+                case req_body_str:
+                    LOGGER.debug("Add in request body: " + rule.getMatchString() + " : " + rule.getReplacement());
+                    String body = msg.getRequestBody().toString();
+                    if (contains(body, rule.getMatchString(), p)) {
+                        body = replace(body, rule.getMatchString(), p, rule.getReplacement());
+                        msg.getRequestBody().setBody(body);
+                        msg.getRequestHeader().setContentLength(msg.getRequestBody().length());
+                    }
+                    break;
+                case resp_header:

```

Comment:
This and following case could be "merged" with `resp_body_str`, it's doing the same thing.